src/azureServiceExplorer/editors/RemoteFileEditor.ts

```
@@ -0,0 +1,68 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { BaseEditor } from '../../azureServiceExplorer/editors/baseEditor';
+import { TextDocument, window } from 'vscode';
+import { TemporaryFile } from '../../components/temporaryFile';
+import * as path from "path";
+import DialogOptions from '../../azureServiceExplorer/messageItems/dialogOptions';
+import * as vscode from "vscode";
+import { IRemoteFileHandler } from './IRemoteFileHandler';
+
+export class RemoteFileEditor<ContextT> extends BaseEditor<ContextT> {
+    constructor(readonly remoteFileHandler:IRemoteFileHandler<ContextT>, readonly showSavePromptKey: string, readonly outputChanel?: vscode.OutputChannel) {
+        super(showSavePromptKey)
+    }
+
+    async getSaveConfirmationText(context: ContextT): Promise<string> {
+        return await this.remoteFileHandler.getSaveConfirmationText(context);
+    }
+
+    async showEditor(context: ContextT): Promise<void> {
+        var fileName = await this.remoteFileHandler.getFilename(context);
+
+        this.appendLineToOutput(`Opening '${fileName}' ...`);
+        try
+        {
+            let parsedPath: path.ParsedPath  =  path.posix.parse(fileName);       
+            let temporaryFilePath = await TemporaryFile.create(parsedPath.base);    
+            await this.remoteFileHandler.downloadFile(context, temporaryFilePath);
+            await super.showEditor(context, temporaryFilePath);
+            this.appendLineToOutput(`Successfully opened '${fileName}'`);
+        } catch (error) {
```

You should always let errors bubble up to the main extension's error handling [here](https://github.com/Microsoft/vscode-azurestorage/blob/master/src/azureServiceExplorer/actions/baseActionHandler.ts#L33) so that:
1. Errors are always caught and tracked in telemetry
1. Error handling is consistent throughout the entire extension
1. If you add handling of a specific type of error - the entire extension benefits from that

Nathan already fixed this in the UI package.